ts/src/profiler.ts

```diff
@@ -13,47 +13,339 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import * as path from 'path';
 
-import {AuthenticationConfig, Common, Logger, Service, ServiceConfig} from '../third_party/types/common-types';
+import * as path from 'path';
+import * as pify from 'pify';
+import * as zlib from 'zlib';
 
-import {Config, ProfilerConfig} from './config';
+import {perftools} from '../../proto/profile';
+import {Common, Logger, Service, ServiceObject} from '../third_party/types/common-types';
+import {ProfilerConfig} from './config';
 import {HeapProfiler} from './profilers/heap-profiler';
 import {TimeProfiler} from './profilers/time-profiler';
 
+export const common: Common = require('@google-cloud/common');
+const parseDuration: (str: string) => number = require('parse-duration');
 const pjson = require('../../package.json');
-const common: Common = require('@google-cloud/common');
+const API = 'https://cloudprofiler.googleapis.com/v2';
+const scope = 'https://www.googleapis.com/auth/monitoring.write';
```

Comment:
nit: use `SCOPE` in caps to indicate constant string (or be consistent, and convert `API` to lowercase).

Reply:
Done.